SimpleWAWS/Code/CsmExtensions/CsmResourceGroupExtensions.cs

```diff
@@ -348,16 +349,36 @@ public static partial class CsmManager
         }
         private static bool IsSimpleWaws(CsmWrapper<CsmResourceGroup> csmResourceGroup)
         {
-            return !string.IsNullOrEmpty(csmResourceGroup.name) &&
-                csmResourceGroup.name.StartsWith(Constants.TryResourceGroupPrefix, StringComparison.OrdinalIgnoreCase) &&
+            return IsSimpleWawsResourceName(csmResourceGroup) &&
                 csmResourceGroup.properties.provisioningState == "Succeeded" &&
                 csmResourceGroup.tags != null && !csmResourceGroup.tags.ContainsKey("Bad")
                 && csmResourceGroup.tags.ContainsKey("FunctionsContainerDeployed");
         }
-        private static bool IsJenkinsResource(CsmWrapper<CsmResourceGroup> csmResourceGroup)
+        private static bool IsSimpleWawsResourceName(CsmWrapper<CsmResourceGroup> csmResourceGroup)
         {
             return !string.IsNullOrEmpty(csmResourceGroup.name) &&
-                csmResourceGroup.name.StartsWith(Constants.TryResourceGroupPrefix, StringComparison.OrdinalIgnoreCase) &&
+                csmResourceGroup.name.StartsWith(Constants.TryResourceGroupPrefix, StringComparison.OrdinalIgnoreCase) ;
+        }
+
+        private static bool IsSimpleWawsResourceActive(CsmWrapper<CsmResourceGroup> csmResourceGroup)
+        {
+            try
+            {
+                return IsSimpleWawsResourceName(csmResourceGroup) &&
+                    csmResourceGroup.tags.ContainsKey(Constants.UserId)
+                    && csmResourceGroup.tags.ContainsKey(Constants.StartTime)
+                    && csmResourceGroup.tags.ContainsKey(Constants.LifeTimeInMinutes)
+                    && DateTime.UtcNow > DateTime.Parse(csmResourceGroup.tags[Constants.StartTime]).AddMinutes(Int32.Parse(csmResourceGroup.tags[Constants.LifeTimeInMinutes]));
+            }
+            catch {
+                //Assume resoourcegroup is in a bad state.
```

Comment:
Okay will log it. Yes. it'll only be a `nullref` here pointing to resources that didn't have any of the expected tags set or highly unlikely the tags were malformed and we get a `FormatException`